lib/formats/sh.c.in

```diff
@@ -269,23 +269,23 @@ static void rm_fmt_head(RmSession *session, RmFmtHandler *parent, FILE *out) {
 
     self->line_count = 0;
     self->session = session;
-    self->user_cmd = rm_fmt_get_config_value(session->formats, "sh", "cmd");
+    self->user_cmd = rm_fmt_get_config_value(session->cfg->formats, "sh", "cmd");
```

Comment:
Sorry for answering late on that. In short: I don't like having globals at all.

> So we get the main disadvantage of a global (unprotected data that can be modified by any function) plus the disadvantage of non-globals (having to pass parameters around).

It's definitely true that ``RmSession`` grew beyond it's health, but that can be solved by passing only limited data to a specific submodule.  IMO the solution is not *"give up, make things global"*, but rather limit the data. You already did a good job on introducing separate "session" objects for e.g. traverse, shredder etc. We could turrn that around in that way that ``RmSession`` contains a ``RmTravSession``. We could only pass ``RmTravSession`` to the traverse module then. This would indirectly lead to more modularized code.

Other reason is: If anyone would like to use ``rmlint`` as library (I know, unlikely, but that's often a good decision base when designing code) then this global state would not work anymore (or just once...). 

I see that it's tempting (and the resulting code actually looks nicer), but I'm not willing to go there. Maybe it's not even that a big of an issue - usually we need a lot of things from ``RmSession`` in each step (``parse``, ``traverse``, ``preprocess``, ``shredder``, ``merge``, ``format``...).